### PR TITLE
Add More SessionWorkflow lifecycle documentation/suggestions

### PR DESCRIPTION
**Pull request by steve-the-edwards**

### CODE REVIEW

File: workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt [hunk 1]
```diff
@@ -70,7 +70,29 @@ public abstract class SessionWorkflow<
    *     or [Flow.onCompletion][kotlinx.coroutines.flow.onCompletion] to handle that.
    *
    *     If you have a general cleanup operation that is fast and thread-safe then you could use
-   *     [Job.invokeOnCompletion][kotlinx.coroutines.Job.invokeOnCompletion].
+   *     [Job.invokeOnCompletion][kotlinx.coroutines.Job.invokeOnCompletion]. Otherwise use
+   *     [awaitCancellation][kotlinx.coroutines.awaitCancellation] with a finally block. Examples:
+   *
+   *     override fun initialState(
```
Comment (rjrjr) [inline workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt:76]: You need to indent this more to get it to format as a code block. Use Ctrl-J to preview.

Comment (rjrjr) [reply workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt:76]: Note that the existing codeblocks are a mess

Comment (steve-the-edwards) [reply workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt:76]: Ok, fixed both.

File: workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt [hunk 2]
```diff
@@ -70,7 +70,29 @@ public abstract class SessionWorkflow<
...
+   *       props: Unit,
+   *       snapshot: Snapshot?,
+   *       workflowScope: CoroutineScope
+   *     ): MyState {
+   *       someService.start()
+   *       workflowScope.coroutineContext.job.invokeOnCompletion {
+   *         // If quick and does not need to on Workflow runtime thread.
```
[the remark this answers is not in the document]
Comment (steve-the-edwards) [reply workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt:83]: "If quick and does not need to execute on the Workflow runtime's thread."

**Review by rjrjr — APPROVED** [review]

**Pull request merged**